### ISSUE TITLE
Brainstorm list of event ideas

### ISSUE THREAD

**Issue by isabela-pf** [isabela-pf/a11y-events#4]
Alt text/image description workshops have been the main focus so far, but we want to come up with other events to improve digital accessibility across the internet. Some of these ideas are to be further developed into events. 

Please note, a11y-events are
- Accessibility-focused. Digital accessibility, really (though I'd be open to hearing other ideas).
- Accessible. These events need to be inclusive, so please make them as accessible as they can be while we are actively remedying inaccessibility elsewhere.
- Reproducible. Single-time events are great, but this project focus on events that can be repeated to help improve accessibility across projects. (Don't let this stop idea sharing; they may be more reproducible than you originally think.)
- Collaborative. These events create community or bring a community along for the ride. While you can do them alone, they need to function when done by a team.
- Informative. These events teach participants so they can carry this accessibility knowledge with them. Lots of inaccessible practices happen purely from unawareness, so these events aim to help.
- Practice-driven. Learning is paired with concrete examples and a chance for participants to try the ideas out for themselves. This practice may help them contribute to a project, not only the event. (Bonus points for practices that do not require participants to manage git or development environments directly.)

**Comment by isabela-pf** [comment]
Some ideas!

- Group/crowdsourced caption files. Produce a .srt file and use github suggestions to have participants add transcription per time stamp. (Attaching these to their videos might be an admin issue, but could be sorted out.)
- Group/crowdsourced manual accessibility testing. Provide a list of tasks to guide people in testing a full tool, a PR, or some other unit. This helps people learn about ways they can review while giving review. They all add their findings via github suggestions, leaving us with a review/testing document.
- ARIA roles. Like the alt text/image description workshops, this proposes adding appropriate labels to applications. The introductory info would need to make it clear that ARIA is additive and we should default to proper HTML first.
- @tonyfast and @gabalafou brought up theming workshops. They were talking about a JupyterLab extension template where everyone could make their own theme, perhaps with different goals for each (ie. a high contrast theme, a grayscale-only theme). So far this sounds Jupyter-specific, but I'd be interested if there were ways we could generalize it for interfaces.
- Table of contents and/or sitemaps. A common and easily-comprehensible accessibility need is having a table of contents or sitemap available, and it often goes overlooked! I'm wondering if there's a way we can set people up to contribute this on documentation or project websites.

**Comment by isabela-pf** [comment]
I'm going to add a few comments with high-level agenda versions of some of the above ideas. I think having them as rough agendas will make them easier to critique (I know they made me question some of my own plans). Then we can keep moving forward with ideas!

I'm adding each as it's own comment so that a) each idea can be directly linked to and b) it helps break them up into (hopefully) clearer parts.

**Comment by isabela-pf** [comment]
## Group/crowdsourced caption files 

End goal: Produce a .srt file to caption a video. (Maybe produce a transcript too?)

Agenda:
- Intro
    - Quick slides about what captions do and don't
    - Intro to .srt file format/relevant styling
    - Participation info
- The main event
    - Go to working PR with a template file divided by relevant timestamps (note to self: check if .srt will preview or this needs to be a plain text file that we change the extension of after)
    - Give people time to choose a section to describe. They can add proposed captions via github suggestions
- Wrap up
    - Identify areas without suggestions (where there will be captions gaps)
    - In early iterations, it was useful to leave time for feedback so I'd want to do that here

**Comment by isabela-pf** [comment]
## Group/crowdsourced manual accessibility testing

End goal: Learn about and practice manual accesibility review in a piece of software. 

(Maybe this will also produce an issue or verify existing issues? This part is still in the air, but I think there could be a way to contribute.)

(I think there are different things that could be reviewed as well. A full piece of software, a subsection, a PR, documentation, so on.)

- Intro
    - Quick slides about manual testing and what we'll do
    - Participation info
- The main event
    - Go to working PR. This will have a markdown document structured with manual tests and relevant links, but with empty lines for the results of those manual tests ([this is a very rough example](https://github.com/isabela-pf/patitos/pull/4/files))
    - Divide up and let people choose a test they want to try. Multiple people can work on the same things if they want
    - They add their thoughts and findings for each test via github suggestions
    - When suggestions are accepted, this creates a document with feedback on the current experience
- Wrap up
    - Identify areas without suggestions (where there will be gaps)
    - (Maybe) compare with existing issues to see if the info fits somewhere?

**Comment by isabela-pf** [comment]
## JupyterLab theming workshops

End goal: Experiment with and produce JupyterLab theme extensions.

(Does this require publishing a package?)

### Idea 1

- Intro
    - Slides about existing JupyterLab themes, visual accessibility summary, and some UI pointers
    - Participation info
- The main event
    - There are multiple, identical PRs (on different branches, but they have the same changes) with a theme extension template. The PR is also annotated with extra doc strings to highlight where participants might want to go to make different changes.
    - Divide participants up to different PRs (give people two minutes to choose if they want to be in a group, then have everyone comment to claim a working PR)
    - People add what theme changes they want via github suggestion
    - There are optional, accessibility-focused prompts for people to follow if they want the inspiration (ie. a high contrast theme, a grayscale-only theme)
- Wrap up
    - Clarify next steps for participants (we'd need to iron this out)
    - Is there a quick way to test/demo people's work? (I'm not sure how feasible it is for these extensions to just immediately be useable)


### Idea 2

- Intro
    - Slides about existing JupyterLab themes, visual accessibility summary, and some UI pointers
    - Participation info
- The main event
    - Have a few pre-made environments for participants to be in JupyterLab (Binder or JupyterLite?)
    - Divide participants up to different PRs (give people two minutes to choose if they want to be in a group, then have everyone comment to claim a working PR)
    - Participants make changes to a template theming notebook and can run all cells to see their changes real time.
    - These notebooks are then saved/contributed somehow (I don't have this part worked out, but there are a few options, including hosts popping in to save the notebooks themselves)
    - There are optional, accessibility-focused prompts for people to follow if they want the inspiration (ie. a high contrast theme, a grayscale-only theme)
    - The contributed notebooks can (somehow) be applied to a template to make a theme
    - Have particpants choose who want to show off their work
- Wrap up
    - Short demos of each group's theme (in their environment)
    - Clarify next steps for participants

**Comment by isabela-pf** [comment]
## Table of contents and/or sitemaps 

End goal: Contribute or update existing table of contents (for documentation) or sitemaps (more broadly) to project websites.

- Intro
    - Quick slides about what sitemaps are and some best practices
    - Participation info
- The main event
    - Go to working PR. This will have placeholders in the diff for where changes should go and directions for how to review the site
    - Participants divide up by section of website to focus on
    - Propose what should be in the sitemap via github suggestions to the skeleton structure. This will mostly be adding links  with proper descriptions and grouping to the template list.
- Wrap up
    - Maybe accept all suggestions and test what they work looks like in action
    - Clarify next steps

**Comment by trallard** [comment]
I like these ideas - shall we transform this into a GH discussion?

**Comment by trallard** [comment]
✅  done and x-linked here for transparency and acknowledgment
